### ISSUE TITLE
IncosistentReferenceDefinition wrong check result

### ISSUE THREAD

**Issue by Nuranto** [magento/magento2#32309]
### Preconditions (*)

Magento 2.4.1

In Magento\Framework\Setup\Declaration\Schema\Declaration\ValidationRules\ IncosistentReferenceDefinition

```
if ($column->getType() !== $referenceColumn->getType()) {
            return true;
        }
```

should be 

```
if ($column->getType() !== $referenceColumn->getType()) {
            return false;
}
``` 

Also, The class is named wrong :) (Incosistent instead of Inconsistent)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by victortodoran** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Oscar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-41642` were created

**Issue Available**: @engcom-Oscar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by gabrieldagama** [comment]
Hi @Nuranto. Thank you for your report. 
The issue has been fixed in magento/magento2#32959 by @engcom-Foxtrot in 2.4-develop branch
Related commit(s):
 - [3d14e8c861b69988189c0da0c29912b59ba93ea1](https://github.com/magento/magento2/commit/3d14e8c861b69988189c0da0c29912b59ba93ea1)
 - [00baa67b7fd039d1f3cee977975f3be583163b37](https://github.com/magento/magento2/commit/00baa67b7fd039d1f3cee977975f3be583163b37)
 - [d98d958d8e306e923160ca03f9d7c1cafa915276](https://github.com/magento/magento2/commit/d98d958d8e306e923160ca03f9d7c1cafa915276)
 - [0ef6ef96cbf44af7f5cfc899d1519d9a0449c2ab](https://github.com/magento/magento2/commit/0ef6ef96cbf44af7f5cfc899d1519d9a0449c2ab)
 - [f9dd81683216e5563620017a611195f3ce21f91d](https://github.com/magento/magento2/commit/f9dd81683216e5563620017a611195f3ce21f91d)
 - [e405b8f476b55e424287b55ab0becaeaa618408a](https://github.com/magento/magento2/commit/e405b8f476b55e424287b55ab0becaeaa618408a)

The fix will be available with the upcoming 2.4.3 release.
